### PR TITLE
Asciidoctor: Copy inline images in lists

### PR DESCRIPTION
**Pull request by nik9000**
Adds support for copying inline images declared in lists. Sadly,
Asciidoctor doesn't make this easy. They pre-convert lists so we have to
detect the image in docbook. I feel like it might have been a mistake to
try and copy images by traversing the tree, mostly because Asciidoctor
works in strange ways.

### CODE REVIEW

**Comment by nik9000** [comment]
I noticed this when working on converting Kibana's book to Asciidoctor.

File: resources/asciidoctor/lib/copy_images/extension.rb [hunk 1]
```diff
@@ -48,12 +49,17 @@ def process_block_image(block)
     end
 
     def process_inline_image(block)
-      return unless block.content_model == :simple
+      process_inline_image_from_source block, block.source if block.content_model == :simple
+      process_inline_image_from_converted block, block.text if
+        block.context == :list_item && block.parent.context == :olist
+    end
```
Comment (estolfo) [inline resources/asciidoctor/lib/copy_images/extension.rb]: You might want to reverse the syntax and write the `if` statement first.
I typically only use the post-fix `if` statement if it fits on one line and is fairly simple logic.

File: resources/asciidoctor/lib/copy_images/extension.rb [hunk 2]
```diff
@@ -68,6 +74,22 @@ def process_inline_image(block)
       end
     end
 
+    ##
+    # Scan the inline image from the generated docbook. It is not nice that
+    # this is required there isn't much we can do about it. We *could* rewrite
```
Comment (estolfo) [inline resources/asciidoctor/lib/copy_images/extension.rb]: Maybe missing the word 'so'..?

File: resources/asciidoctor/lib/copy_images/extension.rb (hunk 1 above)
Comment (nik9000) [reply resources/asciidoctor/lib/copy_images/extension.rb]: Got it!

**Comment by nik9000** [comment]
@esdocs, I've fixed this up from your comments and reworked it to be a bit more consistent with itself.

**Comment by nik9000** [comment]
@estolfo, I've fixed this up from your comments and reworked it a big to be more consistent. Could you have another look?

**Comment by nik9000** [comment]
Thanks so much for reviewing @estolfo!

**Pull request merged**